src/threadnetwork.cs

```diff
@@ -0,0 +1,74 @@
+using CoreFoundation;
+using ObjCRuntime;
+using Foundation;
+
+using System;
+
+namespace ThreadNetwork {
+
+	[iOS (15,0)]
+	[BaseType (typeof (NSObject))]
+	interface THClient
+	{
+		[Async]
+		[Export ("retrieveAllCredentials:")]
+		void RetrieveAllCredentials (Action<NSSet<THCredentials>, NSError> completion);
+
+		[Async]
+		[Export ("deleteCredentialsForBorderAgent:completion:")]
+		void DeleteCredentialsForBorderAgent (NSData borderAgentId, Action<NSError> completion);
```

Comment:
remove `ForBorderAgent` since it's the first parameter, not the action

Reply:
I don't think so, there is a reason for the 'ForBorderAgent'. We have another method called 'RetrieveCredentialsForExtendedPanId' which takes a NsData too, I fear that people will either confuse the NSData between a PanId and a BorderId, so I prefer to be explicit.

Reply:
make sense

I'd usually check the swift names - but it looks like they are not (yet?) on the web doc